varnish-macros/src/lib.rs

```diff
@@ -64,3 +66,33 @@ pub fn vmod(args: pm::TokenStream, input: pm::TokenStream) -> pm::TokenStream {
 
     result.into()
 }
+
+/// Handle the `#[derive(Stats)]` macro. This can only be applied to a struct.
+/// The struct must have only fields of type `AtomicU64`.
+/// - `#[counter]` attribute on a field will export it as a counter.
+/// - `#[gauge]` attribute on a field will export it as a gauge.
+#[proc_macro_derive(Stats, attributes(counter, gauge))]
+pub fn get_metadata(input: pm::TokenStream) -> pm::TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let name = &input.ident;
+    
+    if !stats::has_repr_c(&input) {
```

Comment:
the content of this function should be moved to the `stats.rs` -- this way `proc_macro::TokenStream` (aliased as `pm`) is only used by the compiler to call this function, but the content uses `proc_macro2::*` types -- which allows testing of all this functionality.  Just keep the `parse_macro_input!` here

Reply:
this is still a todo - we have to have a way of testing the output of the proc macro